### PR TITLE
526 itf wrapper

### PR DESCRIPTION
**Pull request by cvalarida**
## ITF success
image
**Context:** The veteran either:
1) Had an active ITF on file already
2) Had no ITFs on file (active, expired, what have you), but the ITF creation succeeded
    - This probably won't happen for increase only, but...I feel like it's safer this way since I don't know if ITFs are removed from the veteran's record after the original claim has gone through

Also shown in this screenshot is where these success alerts will be displayed.

## ITF creation success with old ITF
image
**Context:** The veteran had an old, non-active ITF on file and we were able to submit a new one.

image

## ITF fetch failure
image
**Context:** The request to get the list of ITFs failed.

## ITF creation failure
image
**Context:** Fetching the list of ITFs succeeded, but no active ITF was found, so we tried to create one. This creation request failed.

### CODE REVIEW

**Comment by peggygannon** [comment]
@cvalarida Here are the ITF messages that I provided to @cehsu and @goldenmeanie earlier. Can you let me know if this is the content you're looking for? 

Scenario 1: new application + new Intent to File request
Success message:
**Your Intent to File request has been submitted**
Thank you for submitting your Intent to File request for disability compensation. Your Intent to File will expire on [Month, Day, Year].

Error message:
**Your Intent to File request didn’t go through**
We’re sorry. Your Intent to File request didn’t go through because something went wrong on our end. Please try applying again tomorrow.

Scenario 2: new application + applicant already has an active ITF
**You already have an Intent to File**
Our records show that you already have an Intent to File for disability compensation. Your Intent to File will expire on [Month, Date, Year].

Scenario 3: new application, but unknown ITF status (unknown error, come back tomorrow)
**We can't process your Intent to File request**
We’re sorry. We can't process your Intent to File request at this time. Please try applying again tomorrow.

Scenario 4: returning application, ITF has expired so they need to create a new ITF.

Success message in creating new ITF:
**Your Intent to File request has been submitted**
Thank you for submitting your Intent to File request for disability compensation. Your Intent to File will expire on [Month, Day, Year]

Error message in creating a new ITF:
**Your Intent to File didn’t go through**
We’re sorry. Your Intent to File request didn’t go through because something went wrong on our end. Please try applying again tomorrow.

Scenario 5: returning application and applicant already has an active ITF
**You already have an Intent to File**
Our records show that you already have an Intent to File for disability compensation. Your Intent to File will expire on [Month, Date, Year].

Scenario 6: returning application, but unknown ITF status (unknown error, come back tomorrow)

**We’re sorry. Something went wrong on our end**
We can’t access your Intent to File request right now. Please try applying again tomorrow.

Loading message: 
Please wait while we verify your Intent to File request.

Expander text (for the ITF notifications)
What is an Intent to File request? 

An Intent to File request lets VA know that you’re planning to file a claim. An Intent to File reserves a potential effective date, or the day you could start getting benefits, while you prepare your disability claim and gather supporting documents.

File: src/applications/disability-benefits/526EZ/actions/index.js [hunk 1]
```diff
@@ -1,163 +1,35 @@
...
 
-export const PRESTART_STATUSES = {
-  notAttempted: 'not-attempted',
-  pending: 'pending',
-  succeeded: 'succeeded',
-  failed: 'failed'
-};
+export const ITF_CREATION_INITIATED = 'ITF_CREATION_INITIATED';
+export const ITF_CREATION_SUCCEEDED = 'ITF_CREATION_SUCCEEDED';
+export const ITF_CREATION_FAILED = 'ITF_CREATION_FAILED';
```
Comment (markgreenburg) [inline src/applications/disability-benefits/526EZ/actions/index.js]: Thank you, these actions are super clear.

File: src/applications/disability-benefits/526EZ/components/FormStartControls.jsx [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
 
-import { ITFErrorAlert, VerifiedAlert, UnauthenticatedAlert, UnverifiedAlert, ITFDescription } from '../helpers';
+import { VerifiedAlert, UnauthenticatedAlert, UnverifiedAlert } from '../helpers';
 
 export default function FormStartControls(props) {
   const { user } = props;
 
-  const somethingWentWrong = props.ITFStatus && props.ITFStatus === 'expired' || // This may not be possible
-                             props.errors && props.errors.length;
+  const somethingWentWrong = props.errors && props.errors.length;
```
Comment (markgreenburg) [inline src/applications/disability-benefits/526EZ/components/FormStartControls.jsx]: Should we add `errors` and `user` (used below) to proptypes?

File: src/applications/disability-benefits/526EZ/constants.js [hunk 3]
```diff
@@ -84,3 +86,12 @@ export const ADDRESS_TYPES = {
   mailingAddress: 'mailingAddress',
   forwardingAddress: 'forwardingAddress'
 };
+
+export const itfStatuses = new Proxy({
+  noneFound: 'none found',
```
Comment (markgreenburg) [inline src/applications/disability-benefits/526EZ/constants.js]: is this meant to be snake cased?

**Comment by cehsu** [comment]
@goldenmeanie indicated that the ITF message should appear above the form fields similar to how the prefill message is displayed.

File: src/applications/disability-benefits/526EZ/containers/ITFWrapper.jsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+    // If we've already fetched the ITFs, have none active, and haven't already called createITF, submit a new ITF
+    const hasActiveITF = itf.currentITF && itf.currentITF.status === itfStatuses.active;
+    const createITFCalled = itf.creationCallState !== requestStates.notCalled;
+    if (itf.fetchCallState === requestStates.succeeded && !hasActiveITF && !createITFCalled) {
+      nextProps.createITF();
+    }
+
+    // If we've already displayed the itf success banner, toggle it off when the location changes
+    if (hasActiveITF && !this.state.hasDisplayedSuccess
```
Comment (markgreenburg) [inline src/applications/disability-benefits/526EZ/containers/ITFWrapper.jsx]: Hard to tell for sure, but does this set off any race conditions or other state wonkiness with regard to reliably displaying the success banner exactly one time?

**Review by markgreenburg — APPROVED** [review]
Left some comments but nothing major. LGTM!

**Comment by cvalarida** [comment]
@peggygannon Thanks for that! That's what I'm looking for. I've simplified the scenarios and pulled what seems to be the appropriate content. Could you take a look?

## Fetch success
image

## Submit success (with old ITF)
image
The expanded bit was from @goldenmeanie.

## Fetch failure
image

## Submit failure
image

File: src/applications/disability-benefits/526EZ/tests/containers/ITFWrapper.unit.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    const tree = shallow(
+      <ITFWrapper {...props}>
+        <p>Shouldn't see me yet...</p>
+      </ITFWrapper>
+    );
+    expect(tree.find('AlertBox').length).to.equal(1);
+  });
+
+
+  it('should not submit a new ITF if the fetch failed', () => {
```
Comment (markgreenburg) [inline src/applications/disability-benefits/526EZ/tests/containers/ITFWrapper.unit.spec.jsx]: Good call...this seems pretty important

File: src/applications/disability-benefits/526EZ/components/FormStartControls.jsx [hunk 6]
```diff
@@ -21,20 +19,17 @@ export default function FormStartControls(props) {
         {UnverifiedAlert}
         <a href={`/verify?next=${window.location.pathname}`} className="usa-button-primary verify-link">Verify Your Identity</a>
       </div>}
-      {somethingWentWrong && ITFErrorAlert}
-      {props.ITFStatus === 'pending' && <LoadingIndicator message="Please wait while we check that your intent to file has been submitted."/>}
-      {!props.ITFStatus && user.profile.verified && <SaveInProgressIntro
+      {somethingWentWrong}
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/components/FormStartControls.jsx]: Oh, uh...Yeah, I didn't really think about this much. Looks like we don't need this. :grimacing: Good catch.

File: src/applications/disability-benefits/526EZ/constants.js (hunk 3 above)
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/constants.js]: Oh, this should actually be deleted; it's not returned from the API and we'll know if we have no ITFs by whether the `intentToFile` array is empty.

File: src/applications/disability-benefits/526EZ/containers/ITFWrapper.jsx (hunk 4 above)
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/containers/ITFWrapper.jsx]: I haven't found any. Running through the lifecycle methods in my head, I couldn't figure any way to do break it.

**Review by peggygannon — CHANGES_REQUESTED** [review]
@cvalarida Looks good. I have a few edits to the expanded text that Alex wrote. I'm not sure the question "not what you expected" works here, as people might not even remember that they had an earlier ITF. 

Could we try this instead? 
**Please note:** We found a previous Intent to File request in our records that expired on [month, day, year]. This ITF might have been from an application you started, but didn't finish before the ITF expired. Or, it could have been from a claim you already submitted.

**Review by peggygannon — APPROVED** [review]

**Pull request merged**